marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -618,18 +605,10 @@ shrinkTimeInterval (start, end) =
       , (end  , end  )
       ]
 
-instance SemiArbitrary TimeInterval where
-  semiArbitrary context =
-    do
-      POSIXTime start <- semiArbitrary context
-      duration <- arbitraryNonnegativeInteger
-      pure (POSIXTime start, POSIXTime $ start + duration)
-  semiShrink = fmap shrinkTimeInterval
-
-
 instance Arbitrary ChoiceId where
   arbitrary = ChoiceId <$> arbitraryChoiceName <*> arbitrary
-  shrink (ChoiceId n p) = [ChoiceId n' p' | n' <- shrinkChoiceName n, p' <- shrink p]
+  shrink (ChoiceId n p) = (ChoiceId <$> shrinkChoiceName n <*> pure p)
+                       ++ (ChoiceId n <$> shrink p)
```

Comment:
Improper `shrink` implementation - shrink parts of products independently rather than taking cartesian product of shrinks.